File: frontend/src/api/relations.js
/**
 * GET /api/claims/relations?topic_id=
 * Fetch claim-to-claim relationship graph for a topic.
 */
const BASE_URL = "http://localhost:8000/api";

export async function fetchRelations(topicId) {
  const res = await fetch(`${BASE_URL}/claims/relations?topic_id=${topicId}`);
  if (res.status === 404) {
    throw new Error("404: Session expired or not found.");
  }
  if (!res.ok) throw new Error(`Relations fetch failed: ${res.status}`);
  return res.json();
}

File: backend/api/routes/relations.py
"""
RSE — Relations Route

GET /api/claims/relations?topic_id=
Returns the claim-to-claim relationship graph for a topic.
"""

from fastapi import APIRouter, Query, HTTPException
from services import relation_service

router = APIRouter()


@router.get("/claims/relations")
async def get_relations(topic_id: str = Query(..., description="Topic ID from a previous query")):
    """
    Fetch the relationship graph for a given topic.

    Returns a list of edges: { source, target, type }
    where type is one of: "supports", "contradicts", "related_to"
    """
    relations = await relation_service.get_relations(topic_id)

    if not relations:
        raise HTTPException(
            status_code=404,
            detail=f"No relations found for topic '{topic_id}'. Submit a query first.",
        )

    return relations
